### PR TITLE
build: upgrade Go 1.17 and k8s v0.23.5

### PR DESCRIPTION
**Pull request by xcoulon**
Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

**Review by mfrancisc — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -1,27 +1,98 @@
...
+	k8s.io/kube-openapi v0.0.0-20211115234752-e816edb12b65 // indirect
+	k8s.io/kubectl v0.23.5 // indirect
+	k8s.io/utils v0.0.0-20220210201930-3a6ce19ff2f9 // indirect
+	sigs.k8s.io/json v0.0.0-20211208200746-9f7c6b3444d2 // indirect
+	sigs.k8s.io/structured-merge-diff/v4 v4.2.1 // indirect
+	sigs.k8s.io/yaml v1.3.0 // indirect
 )
 
-go 1.16
+replace github.com/codeready-toolchain/api => github.com/xcoulon/api v0.0.0-20220929075600-5280de04d83f
```
Comment (MatousJobanek) [inline go.mod:98]: reminder

**Review by MatousJobanek — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (xcoulon) [reply go.mod:98]: thanks, but can't really forget since the `ci-check-gomod / go.mod replacements` job failed ;)

Comment (MatousJobanek) [reply go.mod:98]: you know - old habits :-)

Comment (xcoulon) [reply go.mod]: fixed in https://github.com/codeready-toolchain/toolchain-common/pull/271/commits/f75df3d971d4819216bad7e1a53887cacb5d575d

**Pull request merged**